service.py

```
@@ -1,116 +1,420 @@
-            r.raise_for_status()
-        except requests.exceptions.HTTPError as error:
-            print(error)
+            response.raise_for_status()
+            
+            result = response.json()
+            logger.info(f"Transfer successful: {result}")
+            return result
+            
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Transfer failed: {e}")
+            raise APIException(f"Transfer failed: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)